src/routes/AuthLayoutRouter/AuthLayoutRouter.js

```diff
@@ -413,105 +252,115 @@ export default class PrimaryLayout extends Component {
-        {index > 0 && (
-          <button styleName='backButton' {...backProps}>
-            Back
-          </button>
-        )}
-        {continuous && (
-          <button styleName='nextButton' {...primaryProps}>
-            Next
-          </button>
-        )}
-        {!continuous && (
-          <button {...closeProps}>
-            Close
-          </button>
-        )}
-      </div>
-    </div>
+    <Route
+      path={[
+        // In order of more specific to less specific
+        `/:context(all|public)/:view(events|explore|map|projects|stream)/${OPTIONAL_POST_MATCH}`,
+        `/:context(public)/:view(group)/${OPTIONAL_GROUP_MATCH}`,
+        `/:context(all|public)/:view(map)/${OPTIONAL_GROUP_MATCH}`,
+        `/:context(all|public)/${OPTIONAL_POST_MATCH}`,
+        '/:context(all)/:view(topics)/:topicName',
+        '/:context(all)/:view(topics)',
+        `/:context(groups)/:groupSlug/:view(members)/:personId/${OPTIONAL_POST_MATCH}`,
+        `/:context(groups)/:groupSlug/:view(topics)/:topicName/${OPTIONAL_POST_MATCH}`,
+        `/:context(groups)/:groupSlug/:view(map)/${OPTIONAL_GROUP_MATCH}`,
+        `/:context(groups)/:groupSlug/:view(events|explore|groups|map|members|projects|settings|stream|topics)/${OPTIONAL_POST_MATCH}`,
+        `/:context(groups)/:groupSlug/${OPTIONAL_POST_MATCH}`,
+        `/:view(members)/:personId/${OPTIONAL_POST_MATCH}`,
+        '/messages',
+        '/settings',
+        '/search',
+        '/confirm-group-delete'
+      ]}
+      component={routeProps => (
+        <Drawer {...routeProps} styleName={cx('drawer')} {...{ group }} />
+      )}
+    />
   )
 }
+
+const detailRoutes = [
+  { path: `/:context(all|public)/:view(events|explore|map|projects|stream)/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(all|public)/:view(map)/${GROUP_DETAIL_MATCH}`, component: GroupDetail },
+  { path: `/:context(public)/:view(groups)/${GROUP_DETAIL_MATCH}`, component: GroupDetail },
+  { path: `/:context(all)/:view(members)/:personId/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(all|public)/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(groups)/:groupSlug/:view(events|explore|map|projects|stream)/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(groups)/:groupSlug/:view(members)/:personId/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(groups)/:groupSlug/:view(explore|map|groups)/${GROUP_DETAIL_MATCH}`, component: GroupDetail },
+  { path: `/:context(groups)/:groupSlug/:view(topics)/:topicName/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(groups)/:groupSlug/${POST_DETAIL_MATCH}`, component: PostDetail },
+  { path: `/:context(groups)/:groupSlug/${GROUP_DETAIL_MATCH}`, component: GroupDetail },
+  { path: `/:view(members)/:personId/${POST_DETAIL_MATCH}`, component: PostDetail }
+]
+
+// Redirects for legacy and NonAuth routes
+const redirectRoutes = [
```

Comment:
i wonder if/when we could get rid of a lot of these redirects?

Reply:
Yes please! I see no real reason not to. Worst case we'll find a notification email that still has an old URL in it, but that's unlikely at this point. I think any notification emails or bookmarks folks had from earlier days would be gone by now.

Reply:
I just did it and along with it re-integrated the `DrawerRouter` routes into the layout. It really isn't the best candidate for splitting things out anyway. Things are pretty tidy now.

Reply:
All gone!